### ISSUE TITLE
extern "C" declarations/definitions not recognised in C++

### ISSUE THREAD

**Issue by Corax26** [universal-ctags/ctags#931]
In C++, `extern "C"` declarations and definitions are not recognised, except when an `extern "C"` scope is used:

```c++
extern "C" void f(void);
extern "C" void g(void) {}

extern "C"
{
  void h(void);
}
```
> `ctags --c++-kinds=+p`

Output:
```
h	test.cpp	/^  void h(void);$/;"	p	typeref:typename:void	file:
```

This is a regression, compare with the output of ctags 5.8:
```
f	test.cpp	/^extern "C" void f(void);$/;"	p	file:
g	test.cpp	/^extern "C" void g(void) {}$/;"	f
h	test.cpp	/^  void h(void);$/;"	p	file:
```

**Comment by pragmaware** [comment]
Fixed via #937

**Comment by Corax26** [comment]
Great thanks!
